### PR TITLE
[TAO-0000][Bugfix] Fix Video-CLIP skill contracts

### PR DESCRIPTION
**Pull request by lianqiann**
## Summary

- Pin Video-CLIP PyTorch actions to the validated `py3-06` FC image with PyAV 17.1.0, ONNXScript 0.7.1, and no Decord.
- Add TAO Deploy contracts and templates for TensorRT engine generation, evaluation, and inference.
- Restore symbolic ONNX batching with `export.batch_size: -1`; positive values remain documented as fixed-batch exports.
- Correct skill-eval staging to use the provisioned `seanlin` profile and `s3://computex/skill-eval-ci/iv2clip/*`, removing invalid empty fallback prefixes.
- Document online Hugging Face resolution and complete offline S3/local snapshot usage.
- Add focused regression tests for the release, staging, export, and deploy contracts.
- Add a user-facing deploy-image preflight before artifact staging or GPU work.

## Why this is a root-cause fix

The broken behavior came from incomplete and mutually inconsistent release contracts: the PyTorch/deploy action split, release image pins, export shape semantics, decoder dependencies, and deploy templates were not represented together. This change corrects those authoritative contracts and guards them with cross-file regression tests. It does not install packages at runtime, build a derived image, special-case the smoke dataset, or mask import/action failures; those shortcuts would leave new-machine and TensorRT workflows dependent on accidental environment state.

## Tracking

- NVBug 6664615
- NVBug 6664728
- NVBug 6664694
- NVBug 6670029
- Related TAO-PyTorch PR: https://github.com/NVIDIA-TAO/tao-pytorch/pull/114

## Validated images

PyTorch: `nvcr.io/nvstaging/tao/tao-toolkit-pyt:v7.0.1-pyt2.1.0-py3-06`

OCI digest: `sha256:faeb58559e1d87afd16453580999c178feb345f9dc87e35b8f40098e9604dd09`

Deploy: `nvcr.io/nvstaging/tao/tao-toolkit-deploy:7.2.0-rc-47-multiarch`

OCI digest: `sha256:64393cb84b2fde72d2fac819fce216ce863337b533a76303c88fe9ecd1f2c118`

## Testing

- `python -m pytest -q scripts/tests/test_video_clip_release_contracts.py scripts/tests/test_repository_urls.py` — 5 passed
- `bash scripts/validate-skills.sh` — passed
- `python scripts/stamp_versions.py --check` — passed
- `git diff --check` — passed
- PyTorch container smoke — Video-CLIP import passed; PyAV 17.1.0 and ONNXScript 0.7.1 present; Decord absent
- Deploy container smoke against the exact pinned digest — `video_clip gen_trt_engine --help`, `video_clip evaluate --help`, and `video_clip inference --help` each exited 0

The host does not provide `s5cmd`, so live bucket listing remains an eval-environment check.

### CODE REVIEW

File: versions.yaml [hunk 1]
```diff
@@ -38,8 +38,9 @@ images:
     deft_cosmos_reason: nvcr.io/nvidia/tao/tao-toolkit:7.0.1-cosmos-rl
     cosmos_predict:     nvcr.io/nvidia/tao/tao-toolkit:7.0.1-cosmos-predict
     cosmos_embed:       nvcr.io/nvidia/tao/tao-toolkit:7.1.0-cosmos-embed
-    # Interim until tao-pytorch !620 merges and TAO publishes an official IV2CLIP image.
-    video_clip:         nvcr.io/nvstaging/tao/tao-toolkit-pyt:v7.0.1-pyt2.1.0-py3-03
+    # Video-CLIP uses release-candidate images until the TAO 7.2 artifacts reach GA.
+    video_clip:         nvcr.io/nvstaging/tao/tao-toolkit-pyt:v7.0.1-pyt2.1.0-py3-06
+    video_clip_deploy:  nvcr.io/nvstaging/tao/tao-toolkit-deploy:7.2.0-rc-47-multiarch
```
Comment (vpraveen-nv) [inline versions.yaml:45]: The Testing section smoke-tests only the PyTorch image (PyAV/ONNXScript/decord) — was the deploy image itself verified to ship the `video_clip` deploy entrypoint with `gen_trt_engine`/`evaluate`/`inference`? rc pins have burned us before (NVBug 6664626: rc-47 pytorch image shipped the wrong onnxscript despite the pin), and this same `7.2.0-rc-47` generation is being pinned here for deploy. Suggest (a) noting the verifying evidence in the PR body, and (b) adding a deploy-image preflight line to SKILL.md alongside the PyAV check, e.g. `docker run --rm "$VIDEO_CLIP_DEPLOY_IMAGE" video_clip gen_trt_engine --help`.

File: skills/models/tao-finetune-video-clip/references/spec_template_export.yaml [hunk 2]
```diff
@@ -22,11 +22,11 @@ model:
 export:
   checkpoint: /results/train/video_clip_latest.pth
   onnx_file: /results/export/internvideo2clip.onnx
-  encoder_type: separate
+  encoder_type: combined
   input_height: 224
   input_width: 224
   input_channel: 3
-  batch_size: 1
+  batch_size: -1
```
Comment (vpraveen-nv) [inline skills/models/tao-finetune-video-clip/references/spec_template_export.yaml:29]: Collision: #201 modifies this same file (same `batch_size: -1` change, but without the `encoder_type: combined` flip this PR carries). Whichever merges second conflicts. Since this PR owns the full Video-CLIP deploy story and the encoder_type change, suggest #201 drops its hunk on this file and this PR carries the template — worth a quick sync with @ramanathan831.

**Review by vpraveen-nv — APPROVED** [review]

File: versions.yaml (hunk 1 above)
Comment (lianqiann) [reply versions.yaml:45]: Got it @vpraveen-nv  I am not sure if the tao-deploy in skill is required. @christinayyw do you have any suggestions?

Comment (christinayyw) [reply versions.yaml:45]: @lianqiann not sure what the ask is, but if your skill depends on the deploy image I would agree with (b) preflight suggested by @vpraveen-nv

Comment (ramanathan831) [reply versions.yaml:45]: Addressed in 14f5869. I pulled the exact pinned deploy image (digest sha256:64393cb84b2fde72d2fac819fce216ce863337b533a76303c88fe9ecd1f2c118) and verified `video_clip gen_trt_engine --help`, `video_clip evaluate --help`, and `video_clip inference --help`; all exited 0. SKILL.md now performs the requested deploy preflight, the regression suite requires it, and the PR body records the evidence.

File: skills/models/tao-finetune-video-clip/references/spec_template_export.yaml (hunk 2 above)
Comment (ramanathan831) [reply skills/models/tao-finetune-video-clip/references/spec_template_export.yaml:29]: Resolved while incorporating current main in 14f5869. The final template keeps this PR’s complete deploy contract: `encoder_type: combined` and symbolic `batch_size: -1`. The overlapping earlier change is already present on main, and no duplicate/conflict markers remain.

**Comment by christinayyw** [comment]
## TAO Skill Execution Eval Summary

**Latest run:** [tao-sb-OSS-develop #390](https://prod.blsm.nvidia.com/metropolis-iva-tlt-jenkins/job/tao-sb-OSS-develop/390/) — `14f58695` @ 2026-08-29 02:20 UTC

| Skill | Eval Case | Backend | Verdict | Report |
|---|---|---|---|---|
| `skills/models/tao-finetune-video-clip` | — | `claude` | ✅ Pass (⚠️ w/ self-correction) | [Latest Eval Report](https://prod.blsm.nvidia.com/metropolis-iva-tlt-jenkins/job/tao-sb-OSS-develop/390/artifact/jenkins-eval-reports/skills-models-tao-finetune-video-clip-claude/skill-eval-report.html) |
| `skills/models/tao-finetune-video-clip` | — | `codex` | ✅ Pass (⚠️ w/ self-correction) | [Latest Eval Report](https://prod.blsm.nvidia.com/metropolis-iva-tlt-jenkins/job/tao-sb-OSS-develop/390/artifact/jenkins-eval-reports/skills-models-tao-finetune-video-clip-codex/skill-eval-report.html) |


<sub>✅ Pass = expectation met without self-correction &nbsp;·&nbsp; ✅ Pass (⚠️ w/ self-correction) = expectation met but the agent rewrote / retried &nbsp;·&nbsp; ❌ Fail = expectation not met &nbsp;·&nbsp; ❓ Error = shard didn't produce an eval report</sub>
